featuretools/feature_base/feature_base.py

```diff
@@ -82,18 +82,13 @@ def get_name(self):
             self._name = self.generate_name()
         return self._name
 
-    def get_names(self):
-        if not self._names:
-            self._names = self.generate_names()
-        return self._names
-
     def get_feature_names(self):
         n = self.number_output_features
         if n == 1:
-            names = [self.get_name()]
-        else:
-            names = self.get_names()
-        return names
+            return [self.get_name()]
+        if not self._names:
+            self._names = self.generate_names()
+        return self._names
```

Comment:
I think we can improve this a little more - probably no need to create `n`, and we can set `self._names` for single output features as well:
```
def get_feature_names(self):
    if not self._names:
        if self.number_output_features == 1:
            self._names = [self.get_name()]
        else:
            self._names = self.generate_names()
    return self._names
```